### ISSUE TITLE
btcpay valid, but order match stays pending

### ISSUE THREAD

**Issue by niftyboss1** [CounterpartyXCP/counterparty-core#1653]
In this tx, the buyer (bc1qsteve) matches an existing DEX sell order for 73.771 XCP. 
  https://xchain.io/tx/2725230

Buyer then issues a btcpay (using FreeWallet) and sends BTC to the seller:

(https://xchain.io/tx/4cdfe60f261a506bac5bacb8d3ef6e1a38a115c45b5021203193506c752a89e4)

However the order stays pending and no XCP is delivered!

Update: It seems that even though the btcpay tx is valid, the amount being sent is incorrect: only 0.00671060 BTC is sent, rather than 0.01181074 BTC.  

Not sure if this is a FreeWallet or Counterparty issue.

**Comment by ouziel-slama** [comment]
@niftyboss1 thanks for the report! what version of Counterparty freewallet uses ?

**Comment by ouziel-slama** [comment]
will check closer but with v10 events seems OK:
<img width="1424" alt="Capture d’écran 2024-04-14 à 12 58 20" src="https://github.com/CounterpartyXCP/counterparty-core/assets/4574979/29a5136e-5326-4baa-a155-8f9d05beede9">
http://v10.0.0-rc.1.preview.xcp.dev.s3-website-us-east-1.amazonaws.com/block/839087

**Comment by adamkrellenstein** [comment]
@ouziel-slama right now Freewallet is using v10 (api.counterparty.io)

**Comment by ouziel-slama** [comment]
fwiw hashes from v9 and v10.1.0 are matchine after the concerned block:

<img width="1102" alt="Capture d’écran 2024-04-14 à 17 04 32" src="https://github.com/CounterpartyXCP/counterparty-core/assets/4574979/7ed96840-e186-4e7e-9c81-91ca4942c797">

and https://www.xcp.dev/block/839091

this means that if there is a problem it is not specific to v10..

**Comment by ouziel-slama** [comment]
ok here is what I can say for now:

- there is indeed a bug with a BTCPay which is valid but which does not trigger any credit or update of the `order-match`. This bug is present in v9 and v10.1.0
- this bug is not present in v10.0.0-rc.1! (see screenshot of my previous comment)

This means that this bug probably comes from this PR: https://github.com/CounterpartyXCP/counterparty-core/pull/1633 but I still can't understand how.
In any case I don't know if we want to resolve this bug for v10.1.0 because that would diverge from version v9..

**Comment by adamkrellenstein** [comment]
@ouziel-slama I don't follow... is there a consensus break between v9.61.x and v10.0.y?

**Comment by niftyboss1** [comment]
Thanks for the ongoing investigation!

In the meantime I was able to identify and contact the seller, who kindly agreed to send the BTC back (~$430!) to the buyer.   

Still wondering if this could be due to a FreeWallet-specific bug (constructing the btcpay tx with an incorrect amount)?

**Comment by ouziel-slama** [comment]
> @ouziel-slama I don't follow... is there a consensus break between v9.61.x and v10.0.y?

yes, there was one with v10.0.0-rc.1 but it's ok with 10.1.0 (was fixed with https://github.com/CounterpartyXCP/counterparty-core/pull/1633).

**Comment by adamkrellenstein** [comment]
ah okay I didn't realize it was the same issue!

**Comment by ouziel-slama** [comment]
Closing: this is a wallet-side issue (FreeWallet constructed a btcpay with an insufficient BTC amount). Counterparty correctly leaves the order match pending when btc_amount < btc_quantity (BTC must be paid in full), consistent with v9. The only real consensus discrepancy (v10.0.0-rc.1) was already fixed in v10.1.0 via #1633. No protocol change warranted.
